node/node.go

```diff
@@ -388,7 +392,13 @@ func (nc *Config) Build(ctx context.Context) (*Node, error) {
 	//nwork := bsnet.NewFromIpfsHost(innerHost, router)
 	bswap := bitswap.New(ctx, nwork, bs)
 	bservice := bserv.New(bs, bswap)
-	fetcher := net.NewBitswapFetcher(ctx, bservice, blkValid)
+
+	graphsyncNetwork := gsnet.NewFromLibp2pHost(peerHost)
+	bridge := ipldbridge.NewIPLDBridge()
+	loader := gsstoreutil.LoaderForBlockstore(bs)
+	storer := gsstoreutil.StorerForBlockstore(bs)
+	gsync := graphsync.New(ctx, graphsyncNetwork, bridge, loader, storer)
+	fetcher := net.NewGraphSyncFetcher(ctx, gsync, bs, blkValid, peerTracker)
```

Comment:
All this DI is appropriate for this function, but it's a lot of detail added to an already too long method. It would be nice if we could expose a fetcher constructor that takes a peerHost, peerTracker and a block store from the net package. I'm not familiar enough with the abstractions to say if this is a bad idea.

Reply:
gonna make a seperate ticket for this. Thanks!

Reply:
Follow-up: I think it would be better not to do this, because the high odds are we are going to find other uses for Graphsync soon, and may want it available on the node, similar to how the block service (backed by Bitswap) is availabe on the node.

Reply:
Just to be clear. I wasn't suggesting we remove this from Node altogether, just that we push the construction of the GraphSyncFetcher dependencies into the net package.